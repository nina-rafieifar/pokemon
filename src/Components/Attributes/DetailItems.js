import { html } from "lit-element";

export const PokemonWeightAndWeight = (text, Max, Min) => html`<div class="pokemon__details_items">
  <span class="size__text">${text}</span>
  <div class="size">
    <span>${Min} - ${Max}</span>
  </div>
</div>`;

export const PokemonResistantAndWeaknesses = (text, ability) => html`<div class="pokemon__details_items">
  <span class="chip__text">${text}</span>
  <div class="chips">
    ${ability.split(",").map(
      (i) => html`<div class="chip chip_${i.toLowerCase()}">
        <span>${i}</span>
      </div>`
    )}
  </div>
</div>`;
