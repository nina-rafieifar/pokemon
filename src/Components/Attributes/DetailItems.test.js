import { PokemonResistantAndWeaknesses, PokemonWeightAndWeight } from "./DetailItems";
import { fixture } from "@open-wc/testing";

test("PokemonWeightAndWeight renders the correct elements", async () => {
  const element = await fixture(PokemonWeightAndWeight("Weight", 10, 12));
  expect(element).toMatchSnapshot();
});

test("PokemonResistantAndWeaknesses renders the correct elements", async () => {
  const element = await fixture(PokemonResistantAndWeaknesses("Weight", "water,fire"));
  expect(element).toMatchSnapshot();
});
