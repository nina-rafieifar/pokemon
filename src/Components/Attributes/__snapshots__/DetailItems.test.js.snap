// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PokemonResistantAndWeaknesses renders the correct elements 1`] = `
<div
  class="pokemon__details_items"
>
  
  
  <span
    class="chip__text"
  >
    <!---->
    Weight
    <!---->
  </span>
  
  
  <div
    class="chips"
  >
    
    
    <!---->
    <div
      class="chip chip_water"
    >
      
        
      <span>
        <!---->
        water
        <!---->
      </span>
      
      
    </div>
    <!---->
    <div
      class="chip chip_fire"
    >
      
        
      <span>
        <!---->
        fire
        <!---->
      </span>
      
      
    </div>
    <!---->
    
  
  </div>
  

</div>
`;

exports[`PokemonWeightAndWeight renders the correct elements 1`] = `
<div
  class="pokemon__details_items"
>
  
  
  <span
    class="size__text"
  >
    <!---->
    Weight
    <!---->
  </span>
  
  
  <div
    class="size"
  >
    
    
    <span>
      <!---->
      12
       - 
      10
      <!---->
    </span>
    
  
  </div>
  

</div>
`;
